spacy_loggers/mlflow.py

```diff
@@ -18,74 +59,131 @@ def mlflow_logger_v1(
+        )
+
+        def log_step(info: Optional[Dict[str, Any]]):
+            console_log_step(info)
+            _log_step_mlflow(mlflow, info)
+
+        def finalize() -> None:
+            console_finalize()
+            _finalize_mlflow(mlflow)
+
+        return log_step, finalize
+
+    return setup_logger
+
+
+def _import_mlflow() -> ModuleType:
     try:
         import mlflow
 
         # test that these are available
-        from mlflow import start_run, log_metric, log_metrics, log_artifact, end_run  # noqa: F401
+        from mlflow import (
+            start_run,
+            log_metric,
+            log_metrics,
+            log_artifact,
+            end_run,
+        )  # noqa: F401
     except ImportError:
         raise ImportError(
             "The 'mlflow' library could not be found - did you install it? "
             "Alternatively, specify the 'ConsoleLogger' in the "
             "'training.logger' config section, instead of the 'MLflowLogger'."
         )
+    return mlflow
 
-    console = console_logger(progress_bar=False)
 
-    def setup_logger(
-        nlp: Language, stdout: IO = sys.stdout, stderr: IO = sys.stderr
-    ) -> Tuple[Callable[[Dict[str, Any]], None], Callable[[], None]]:
-        config = nlp.config.interpolate()
-        config_dot = util.dict_to_dot(config)
-        for field in remove_config_values:
-            del config_dot[field]
-        config = util.dot_to_dict(config_dot)
-        mlflow.start_run(
-            run_id=run_id,
-            experiment_id=experiment_id,
-            run_name=run_name,
-            nested=nested,
-            tags=tags,
-        )
+def _setup_mlflow(
+    mlflow: ModuleType,
+    nlp: Language,
+    run_id: Optional[str] = None,
+    experiment_id: Optional[str] = None,
+    run_name: Optional[str] = None,
+    nested: bool = False,
+    tags: Optional[Dict[str, Any]] = None,
+    remove_config_values: List[str] = SimpleFrozenList(),
+):
+    config = nlp.config.interpolate()
+    config_dot = dict_to_dot(config)
+    for field in remove_config_values:
+        del config_dot[field]
+    config = dot_to_dict(config_dot)
 
-        config_dot_items = list(config_dot.items())
-        config_dot_batches = [
-            config_dot_items[i : i + 100] for i in range(0, len(config_dot_items), 100)
-        ]
-        for batch in config_dot_batches:
-            mlflow.log_params({k.replace("@", ""): v for k, v in batch})
+    mlflow.start_run(
+        run_id=run_id,
+        experiment_id=experiment_id,
+        run_name=run_name,
+        nested=nested,
+        tags=tags,
+    )
 
-        console_log_step, console_finalize = console(nlp, stdout, stderr)
+    config_dot_items = list(config_dot.items())
+    config_dot_batches = [
+        config_dot_items[i : i + 100] for i in range(0, len(config_dot_items), 100)
+    ]
+    for batch in config_dot_batches:
+        mlflow.log_params({k.replace("@", ""): v for k, v in batch})
```

Comment:
Do you know why we log the params in batches?

Reply:
No, I'm afraid not. The original MLflow logger was introduced in a 3rd party PR.